scout/server/blueprints/panels/controllers.py

```diff
@@ -105,7 +105,8 @@ def update_existing_panel(store, request, lines):
             "Permission denied: please ask a panel maintainer or admin for help.",
             "danger",
         )
-    return panel_obj.get("_id")
+    if panel_obj:
```

Comment:
I might be reading this too quickly, but shouldn't this move up into the if granted clause, the else be removed and a None returned after the flash?

Reply:
I works, but I can move it up if you want

Reply:
Its subtly different. Notice the prior return None if panel_obj is None.

Reply:
Lets see, how is that redirect_id used..

Reply:
The problem is not the user not having permissions, but an error in the function `update_panel` due to a malformed gene panel file, but the latest code should cover both eventualities

Reply:
Yea, if the user didn't have permission it wasn't going to crash, since gene_panel exists (checks the line before in the function)